modules/custom/az_ranking/az_ranking.module

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * @file
+ * Contains az_ranking.module.
+ */
+
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\Core\Entity\EntityFormInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+/**
+ * Implements hook_help().
+ */
+function az_ranking_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the az_ranking module.
+    case 'help.page.az_ranking':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Custom field plugins and theme template for rankings.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function az_ranking_theme($existing, $type, $theme, $path) {
+  return [
+    'az_ranking' => [
+      'variables' => [
+        'attributes' => [],
+        'media' => NULL,
+        'column_span' => NULL,
+        'ranking_clickable' => NULL,
+        'ranking_hover_effect' => NULL,
+        'ranking_heading' => NULL,
+        'ranking_alignment' => NULL,
+        'ranking_header_style' => NULL,
+        'ranking_description' => NULL,
+        'ranking_source' => NULL,
+        'ranking_source_classes' => NULL,
+        'ranking_font_color' => NULL,
+        'text_color_override' => NULL,
+        'link' => NULL,
+        'link_url' => NULL,
+        'link_title' => NULL,
+        'ranking_link_style' => NULL,
+      ],
+      'template' => 'az-ranking',
+    ],
+  ];
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function az_ranking_form_alter(&$form, FormStateInterface $form_state, $form_id) {
```

Comment:
Can [hook_form_BASE_FORM_ID_alter](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21form.api.php/function/hook_form_BASE_FORM_ID_alter/11.x) be used instead of `hook_form_alter()`?